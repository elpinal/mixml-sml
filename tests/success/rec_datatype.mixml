link X = {type void : Type} with

new [unit
  link
    link Y = {type void : Type} with
    link {val void_in : ext X.void -> ext Y.void} with
    {val void_out : ext Y.void -> ext X.void}
  seals

  link Z = {type void = ext X.void} with
  link {val void_in = fun (x : ext Z.void) -> x} with
  {val void_out = fun (x : ext Z.void) -> x}
]
