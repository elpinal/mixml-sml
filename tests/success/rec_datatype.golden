{void : [= !0], void_in : [!0 -> !0]+, void_out : [!0 -> !0]+}
